transaction/core/src/input_rules.rs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2022 The MobileCoin Foundation
+
+//! Input rules, described in MCIP #31, specify any additional criteria that the
+//! Tx must satisfy to be valid.
+//!
+//! Input rules make sense when a Tx is built collaboratively, with some inputs
+//! coming from some parties, and some inputs come from others. They give
+//! participants a way to make their signature contingent on certain rules being
+//! followed, to facilitate trustless interactions.
+
+use crate::{
+    tx::{Tx, TxOut},
+    BlockVersion,
+};
+use alloc::vec::Vec;
+use displaydoc::Display;
+use mc_crypto_digestible::{Digestible, MerlinTranscript};
+use prost::Message;
+use serde::{Deserialize, Serialize};
+
+/// A representation of rules on a transaction, imposed by the signer of some
+/// input in the transaction.
+///
+/// Any rule could conceivably be added here if it can be evaluated against a
+/// `Tx`.
+#[derive(Clone, Digestible, PartialEq, Eq, Message, Serialize, Deserialize)]
+pub struct InputRules {
+    /// Outputs that are required to appear in the Tx prefix for the transaction
+    /// to be valid
+    #[prost(message, repeated, tag = "1")]
+    pub required_outputs: Vec<TxOut>,
+
+    /// An upper bound on the tombstone block which must be respected for the
+    /// transaction to be valid
+    #[prost(fixed64, tag = "2")]
+    pub max_tombstone_block: u64,
+}
+
+impl InputRules {
```

Comment:
Are there tests for these input rules? Would be good to have simple unit tests verifying failure cases if not.

Reply:
Yeah -- there are the validation tests which cover this, but we should also test this API I guess